### PR TITLE
Crate level documentation

### PR DESCRIPTION
**Pull request by vccggorski**

### CODE REVIEW

File: mcan/src/lib.rs [hunk 1]
```diff
@@ -1,6 +1,263 @@
...
+//! requirements of [`mcan_core`] traits which platform-specific HALs are
+//! expected to implement.
+//!
+//! In order to use MCAN, one has to instantiate [`CanConfigurable`] and
+//! [`finalize`] it. Its constructor requires an instance of an
+//! [`Dependencies`] implementing struct and holds onto it until it's
+//! [`freed`]. Safety requirements of the `Dependencies` trait
+//! guarantee a correct state of MCAN interfaces during its operation.
+//!
+//! TODO: We do not have `CanConfigurable::free`??
```
Comment (vccnfitingh) [inline mcan/src/lib.rs:37]: I guess we'll just have to add it. Bikeshed the name in #28.

File: mcan/src/lib.rs [hunk 2]
```diff
@@ -1,6 +1,263 @@
...
+//! region provided by a user is sound; yet it is up to the user to put it in a
+//! valid, accessible to MCAN, RAM memory region.
+//!
+//! One can configure the Message RAM as follows
+//! - specify a custom MEMORY entry in a linker script mapped to the valid RAM
+//!   memory region
+//! - introduce a custom, `.bss` like (NOLOAD property), section - eg. `.can`
+//! - map a new custom input section to a new custom MEMORY entry
+//! - use the `#[link_section]` attribute in a code to link a static variable to
+//!   this memory region
```
Comment (vccnfitingh) [inline mcan/src/lib.rs:59]:
```suggestion
//! One can configure the Message RAM as follows
//! - specify a custom `MEMORY` entry in a linker script mapped to the valid RAM
//!   memory region
//! - introduce a custom, `.bss` like (`NOLOAD` property), section - eg. `.can`
//! - map the input section to the `MEMORY` entry
//! - use the `#[link_section]` attribute to link a static variable to
//!   this memory region
```

File: mcan/src/lib.rs [hunk 3]
```diff
@@ -1,6 +1,263 @@
...
+//!
+//! Example of a linker script
+//! ```text
+//! MEMORY
+//! {
+//!   FLASH : ORIGIN = 0x400000, LENGTH = 2M
+//!   CAN : ORIGIN = 0x20400000, LENGTH = 64K
```
Comment (vccnfitingh) [inline mcan/src/lib.rs:66]: One MCAN can use up to 4352 32-bit words (17408 bytes) of RAM. This 64 KiB region is enough for 3 full MCAN peripherals, or 4 in a slightly less extreme configuration. I think the example should show 17 K.

image

**Review by vccnfitingh — CHANGES_REQUESTED** [review]

File: mcan/src/lib.rs (hunk 1 above)
Comment (vccggorski) [reply mcan/src/lib.rs:37]: Done.

File: mcan/src/lib.rs (hunk 2 above)
Comment (vccggorski) [reply mcan/src/lib.rs]: Done

File: mcan/src/lib.rs (hunk 3 above)
Comment (vccggorski) [reply mcan/src/lib.rs:64]: I think this is fine

Comment (vccnfitingh) [reply mcan/src/lib.rs:64]: I guess this is OK since space-constrained users will want to tailor the size anyway.

**Review by vccnfitingh — APPROVED** [review]

**Pull request merged**